### PR TITLE
Appdynamics Collector Widget

### PR DESCRIPTION
**Pull request by revelashaun**
**Features:**
MVP1: Appdynamics Widget
The Appdynamics Collector Polls Data on:

1. Health Rules and their violation statuses for your application (as you've configured them within Appdynamics).  This is broken down by node health and transaction health.  (if you click on the error / warning icons, drill down data is also included.
2. Overall Average Response Time (Transaction-level response times would be nice, but required more website real estate than we could take)
3. Error Rates
4. Throughput

Incomplete: Customizable Dashboards and Widgets

All work performed at the University of Illinois Digital Campus Labs, led by Shaun Dolan (@revelashaun) and Satish Ramabhadran (@satishc1).

A special thanks to Patrick Mackay and @tabladrum for their leadership.

Thank you Eric, Mike, and Austin for being **Relentless Developers**:

1. **Mike Peritz**; @mikeperetz - "Mr. API", he handled a majority of the backend collector work coordinating closely with Topo, me, and @amp5208 
2. **Eric Yamane**; @Eric2991 - Polishing the finishing touches on the Appdynamics Widget UI and also working on a customizable dashboard template framework; In addition he ran assessments for a Splunk Collector, but we didn't feel that it fit into the Widget Theme.
3.  **Austin Bae**; @hyunraider - Core UI Framework for the base Appdynamics Widget
4.  **Nireesh Thiruveedula**; @NireeshT - Thanks for your expert/technical guidance in helping me rebase since we were so far behind master on our branch.  It wouldn't be released today without your help.

Screenshots:
screen shot 2017-04-26 at 12 21 49 pm
screen shot 2017-04-26 at 12 22 02 pm

### CODE REVIEW

**Comment by MarkRx** [comment]
Do you have a description or any screenshots?

**Comment by nireeshT** [comment]
@MarkRx - here you go.  
screen shot 2017-04-12 at 12 52 43 pm

File: .gitignore [hunk 1]
```diff
@@ -38,8 +38,9 @@ logs
...
-.pmdruleset.xml
-.pmd
-.classpath
-application.properties
+*.properties
+/dashboard.properties
+
+/appdynamics-collector/src/main/resources/appdynamics.properties
+/appdynamics-collector/appdynamics.properties
+/api/api.properties
```
Comment (satishc1) [inline .gitignore]: The appdynamics specfic stuff should be within the appdynamics collector .gitignore.

**Comment by revelashaun** [comment]
@satishc1 and @nireeshT - please accept this pull request when ready.  Please let me know if I need to bring anything current, I want to avoid too much code divergence after the last rebase.

Again, thank you both for all of your help.  It's been a learning experience for me.

**Comment by nireeshT** [comment]
@revelashaun  - PR is in review. We will get it merged at the earliest.

**Comment by satishc1** [comment]
@revelashaun @tabladrum @nireeshT LGTM

**Pull request merged**